lib/signer.ex

```diff
@@ -62,13 +62,19 @@ defmodule AntikytheraAws.Signer do
           ad = Time.to_iso_basic(Time.now())
           {ad, Map.put(without_amz_date, "x-amz-date", ad)}
       end
-      sign_ready_headers = if st, do: Map.put(headers_with_date, "x-amz-security-token", st), else: headers_with_date
+      headers_might_with_st = if st, do: Map.put(headers_with_date, "x-amz-security-token", st), else: headers_with_date
+      payload_sha256        = hex_sha256(payload)
+      sign_ready_headers    = if service == "s3" do
+        Map.put(headers_might_with_st, "x-amz-content-sha256", payload_sha256)
+      else
+        headers_might_with_st
+      end
```

Comment:
Could you refactor body of this function into several private functions as it's now too long?

Reply:
Yes of course. 
fixed in 52a8a8b